### PR TITLE
Upgrade OS Version to 2.0.0-alpha1-SNAPSHOT

### PR DESCRIPTION
**Pull request by vamsi-amazon**
Signed-off-by: vamsi-amazon <reddyvam@amazon.com>

### Description
Upgrade OpenSearch version to 2.0.0-alpha1-SNAPSHOT  which has more changes corresponding to removal of mapping types.
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/452
https://github.com/opensearch-project/sql/issues/485
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/sql-test-and-build-workflow.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: SQL Java CI
 on: [push, pull_request]
 
 env:
-  OPENSEARCH_VERSION: '2.0.0-SNAPSHOT'
```
Comment (joshuali925) [inline .github/workflows/sql-test-and-build-workflow.yml:6]: this should stay as 2.0.0-SNAPSHOT, then add another variable for `alpha1`, and update steps?
```yaml
    - name: Build with Gradle
      run: ./gradlew build assemble -Dopensearch.version=${{ env.OPENSEARCH_VERSION }} -Dbuild.version_qualifier=${{ env.QUALIFIER }}
```

File: build.gradle [hunk 2]
```diff
@@ -37,10 +38,15 @@ repositories {
 
 ext {
     isSnapshot = "true" == System.getProperty("build.snapshot", "true")
+
```
Comment (joshuali925) [inline build.gradle:41]: instead of hard coding buildVersionQualifier on top, it should be read from properties in here. if you add this change i can close #516 
```
buildVersionQualifier = System.getProperty("build.version_qualifier")
```

**Review by joshuali925 — COMMENTED** [review]
minor comments. FYI the workbench snapshots change should be ok for us, it's coming from dashboards core EUI library update

File: integ-test/build.gradle [hunk 3]
```diff
@@ -53,8 +53,8 @@ configurations.all {
     // enforce 1.1.3, https://www.whitesourcesoftware.com/vulnerability-database/WS-2019-0379
     resolutionStrategy.force 'commons-codec:commons-codec:1.13'
     resolutionStrategy.force 'com.google.guava:guava:31.0.1-jre'
-    resolutionStrategy.force 'com.fasterxml.jackson.core:jackson-core:2.12.6'
-    resolutionStrategy.force 'com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:2.12.6'
+    resolutionStrategy.force 'com.fasterxml.jackson.core:jackson-core:2.13.2'
+    resolutionStrategy.force 'com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:2.13.2'
```
Comment (penghuo) [inline integ-test/build.gradle:57]: suggest to upgrade to 2.13.2.1?

File: .github/workflows/sql-test-and-build-workflow.yml (hunk 1 above)
Comment (joshuali925) [reply .github/workflows/sql-test-and-build-workflow.yml:6]: ignore this comment, although i saw they still want us to include `-Dbuild.version_qualifier=`

**Review by joshuali925 — DISMISSED** [review]

File: integ-test/build.gradle (hunk 3 above)
Comment (vamsi-amazon) [reply integ-test/build.gradle:57]: Done.

**Review by joshuali925 — APPROVED** [review]

File: legacy/src/main/java/org/opensearch/sql/legacy/esdomain/mapping/FieldMapping.java [hunk 4]
```diff
@@ -118,8 +118,8 @@ public String path() {
      */
     @SuppressWarnings("unchecked")
     public String type() {
-        FieldMappingMetadata metaData = typeMappings.getOrDefault(fieldName, FieldMappingMetadata.NULL);
-        if (metaData.isNull()) {
+        FieldMappingMetadata metaData = typeMappings.get(fieldName);
+        if (metaData == null) {
```
Comment (dai-chen) [inline legacy/src/main/java/org/opensearch/sql/legacy/esdomain/mapping/FieldMapping.java:122]: np: wondering why this change required?

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

**Review by penghuo — APPROVED** [review]
LGTM

File: legacy/src/main/java/org/opensearch/sql/legacy/esdomain/mapping/FieldMapping.java (hunk 4 above)
Comment (vamsi-amazon) [reply legacy/src/main/java/org/opensearch/sql/legacy/esdomain/mapping/FieldMapping.java:122]: FieldMappingMetadata.NULL this static field is removed in the upstream os library.

Comment (dai-chen) [reply legacy/src/main/java/org/opensearch/sql/legacy/esdomain/mapping/FieldMapping.java:122]: got it. thanks!
